oauthenticator/google.py

```diff
@@ -115,7 +115,7 @@ def _cast_hosted_domain(self, proposal):
         if self.hosted_domain:
             if (
                 user_email_domain not in self.hosted_domain
-                or bodyjs['hd'] not in self.hosted_domain
+                or ('hd' in bodyjs and bodyjs['hd'] not in self.hosted_domain)
```

Comment:
Reading in https://developers.google.com/identity/protocols/OpenIDConnect#hd-param I wonder if this logic really is relevant. I see that the hd value can even be an `*` to indicate something entirely different.

I guess that you ran into an issue where the hd parameter wasn't passed, and you ended up with an error of trying to access the `hd` key of bodyjs, and that led to this PR?

Is the right fix perhaps to remove this section? I don't fully grasp what the hd (hosted domain) parameter indicates though.

Reply:
From my reading of the docs you linked and what I know about GSuites I would say we can drop this. It is a hint to the UI of Google which G Suite hosted domain you are looking for.

Reply:
@consideRatio Your guess is correct... removing the requirement for the `hd` value, and just using the `user_email_domain` feels like a cleaner way to filter.

@betatim I'll amend my PR accordingly.

Thanks both!